code/modules/research/xenoarchaeology/artifact/artifact_duplicator.dm

```diff
@@ -0,0 +1,188 @@
+	. = ..()
+	update_icon()
+	if(anchored && !deployed)
+		deploy_pods()
+	else if(!anchored && deployed)
+		retract_pods()
+
+/obj/machinery/duplicator/proc/deploy_pods()
+	desc = "It's some kind of large computer."
+	visible_message("The disks detach from the sides of \the [src] and unfold into large pods.")
+	leftpod.forceMove(locate(x-1,y,z))
+	rightpod.forceMove(locate(x+1,y,z))
+	flick("duplicator_pad_deploying", leftpod)
+	flick("duplicator_pad_deploying", rightpod)
+	deployed = TRUE
+
+/obj/machinery/duplicator/proc/retract_pods()
+	desc = initial(desc)
+	visible_message("The pods retract into their bases and reattach to the sides of \the [src].")
+	leftpod.forceMove(src)
+	rightpod.forceMove(src)
+	deployed = FALSE
+
+/obj/machinery/duplicator/attack_hand(mob/user as mob)
+	if(..() || !anchored)
+		return 1
+	interact(user)
+
+/obj/machinery/duplicator/interact(mob/user)
+	if(!leftpod || !rightpod)
+		visible_message("\The [src] buzzes \"<span class='warning'>ERROR: Both pods not present.</span>\"")
+		flick("duplicator_denied", src)
+		return
+	if(!leftpod.get_object() || !rightpod.get_object())
+		if(leftpod.get_object())
+			if(!leftpod.has_multiple_objects())
+				var/atom/movable/A = leftpod.get_object()
+				if(!A.anchored && istype(A))
+					begin_duplication(leftpod, rightpod, A)
+				else
+					visible_message("\The [src] buzzes \"<span class='warning'>ERROR: Object anchored, pod cannot close.</span>\"")
+					flick("duplicator_denied", src)
+			else
+				visible_message("\The [src] buzzes \"<span class='warning'>ERROR: Too many objects present in pod.</span>\"")
+				flick("duplicator_denied", src)
+		else if(rightpod.get_object())
+			if(!rightpod.has_multiple_objects())
+				var/atom/movable/A = rightpod.get_object()
+				if(!A.anchored && istype(A))
+					begin_duplication(rightpod, leftpod, A)
+				else
+					visible_message("\The [src] buzzes \"<span class='warning'>ERROR: Object anchored, pod cannot close.</span>\"")
+					flick("duplicator_denied", src)
+			else
+				visible_message("\The [src] buzzes \"<span class='warning'>ERROR: Too many objects present in pod.</span>\"")
+				flick("duplicator_denied", src)
+		else
+			visible_message("\The [src] buzzes \"<span class='warning'>ERROR: No object present in pod.</span>\"")
+	else
+		visible_message("\The [src] buzzes \"<span class='warning'>ERROR: Both pods obstructed.</span>\"")
+		flick("duplicator_denied", src)
+
+/obj/machinery/duplicator/proc/begin_duplication(obj/structure/duplicator_pod/sourcepod, obj/structure/duplicator_pod/destpod, atom/movable/A)
+	if(!sourcepod || !destpod || !istype(A))
+		visible_message("\The [src] buzzes \"<span class='warning'>ERROR: Unspecified error. Please contact maintenance.</span>\"")
+		return
+	duplicating = TRUE
+	update_icon()
+	A.forceMove(sourcepod)
+	sourcepod.icon_state = "borgcharger1(old)"
+	sourcepod.density = 1
+	destpod.icon_state = "borgcharger1(old)"
+	destpod.density = 1
+	A.duplicate(destpod)
+	spawn(dupe_time)
```

Comment:
Because there's no code after this spawn, it's better to set `waitfor = 0` at the start of the proc and use a sleep instead - it helps with horizontal readability.